xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/internal/DefaultEntityResolver.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.xml.internal;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.xml.EntityResolver;
+
+/**
+ * Uses css4j's entity resolver.
+ * <p>
+ * That resolver provides the following features:
+ * </p>
+ * <ul>
+ * <li>Security protections against <a href=
+ * "https://owasp.org/www-community/attacks/Server_Side_Request_Forgery">SSRF</a>
+ * (although better security could be achieved by enabling the whitelist) and
+ * {@code jar:} decompression bombs.</li>
+ * <li>Has about all W3C DTDs built-in for fast access.</li>
+ * <li>Enables the usage of the XHTML5 DTD or even DTD-less documents.</li>
+ * <li>Allows reading preset DTDs from the classpath or modulepath.</li>
+ * </ul>
+ *
+ * @version $Id$
+ */
+@Component
+@Singleton
+public class DefaultEntityResolver extends io.sf.carte.doc.xml.dtd.DefaultEntityResolver implements EntityResolver
+{
+    public DefaultEntityResolver() {
+        super(false);
+        // The following is used by org.apache.maven.wagon:wagon-webdav-jackrabbit
```

Comment:
Thanks for the comment but why is that needed? Is it because by default the resolver will only resolve DTDs locally and we don't provide locally the `graffito-jcr-mapping.dtd` needed by `wagon-webdav-jackrabbit`?  I'm curious to understand how it was working before your change. Also, shouldn't we provide it locally to speed it up and avoid the internet lookup? Actually I'm even wondering why we need wagon-webdav-jackrabbit as I doubt that we use it, so we might just want to exclude that dependency altogether (would need to check where it's used).

Thanks

Reply:
> Actually I'm even wondering why we need wagon-webdav-jackrabbit as I doubt that we use it, so we might just want to exclude that dependency altogether (would need to check where it's used).

About this part, I can already confirm that we don't need it for XWiki at runtime since I cannot find it in `WEB-INF/lib`.

Reply:
From the `pom.xml`:

```xml
<plugin>
  <groupId>org.apache.maven.plugins</groupId>
  <artifactId>maven-site-plugin</artifactId>
  <!-- Lock down plugin version for build reproducibility -->
  <version>3.3</version>
  <!-- Site Plugin doesn’t automatically pull the Apache Wagon Dav implementation which is needed for
	   site:deploy goal when using Dav protocol. -->
  <dependencies>
	<dependency>
	  <groupId>org.apache.maven.wagon</groupId>
	  <artifactId>wagon-webdav-jackrabbit</artifactId>
	  <version>2.6</version>
	</dependency>
  </dependencies>
</plugin>
```

Reply:
In any case, `maven-site-plugin` does not use XWiki's resolver, and a `fgrep --directories=recurse jackrabbit *` on XWiki's repositories does not yield anything, so the only reason to not remove the reference to that DTD would be if users are expected to use it in documents.

May I remove it?

Reply:
> In any case, maven-site-plugin does not use XWiki's resolver

I was about to say that :)

> May I remove it?

Yes, I don't see a need ATM.

Thanks

Reply:
Removed.